### DISCUSSION TITLE
Seeing errors when calling List Responses on PullRequest Comment

### DISCUSSION THREAD

**Discussion by myyk** [community/community#26058] [Actions]
<p>Hello,</p>
<p>I am attempting to Reactions on a PullRequest Comment and I’m seeing this “Resource not accessible by integration” error. I’ve seen that people get that error when trying to write when they only have read access. I would think this API only needs read access though.</p>
<p>I believe I’m using <code>GITHUB_TOKEN</code> as my token so I’m a bit confused. Is it possible that this API is setup to need write access? Did I miss something?</p>
<p>My <code>action.yml</code> looks like this:</p>
<pre><code class="lang-auto">author: Myyk
description: Tally votes on PR
inputs:
  token:
    description: 'GITHUB_TOKEN or a repo scoped PAT.'
    default: ${{ github.token }}
  repository:
    description: 'The full name of the repository in which to tally votes.'
  comment-id:
    description: Comment id of comment containing votes.
    required: true
outputs:
  for:
    description: The number for the change.
  against:
    description: The number against the change.
runs:
  using: node12
  main: index.js
</code></pre>
<p>In another action in the same workflow, I’m successfully modifying a PR comment, so I seem to have write access on my token. I’m pretty confused what’s going on here.</p>
<p>I’m creating my octokit like this in my js code following <a href="https://github.com/actions/toolkit/tree/main/packages/github" rel="nofollow noopener">https://github.com/actions/toolkit/tree/main/packages/github</a> since that API needs the preview parameter:</p>
<pre><code class="lang-auto">    const octokit = github.getOctokit(
      inputs.token,
      {
        previews: ['squirrel-girl'],
      },
    );
</code></pre>
<p>Calling the API with something like this:</p>
<pre><code class="lang-auto">      octokit.reactions.listForPullRequestReviewComment({
        owner: owner,
        repo: repo,
        comment_id: commentId,
        content: reaction,
      });
</code></pre>
<p>I’m not sure if the curl works on the endpoint even. Trying to follow the docs here <a href="https://docs.github.com/en/rest/reference/reactions#list-reactions-for-an-issue" rel="nofollow noopener">https://docs.github.com/en/rest/reference/reactions#list-reactions-for-an-issue</a>.</p>
<pre><code class="lang-auto">curl  \
  -H "Accept: application/vnd.github.squirrel-girl-preview+json" \
  https://api.github.com/repos/myyk/git-democracy/pulls/comments/677573350/reactions
</code></pre>
<p>Seeing:</p>
<pre><code class="lang-auto">  "message": "Not Found",
  "documentation_url": "https://docs.github.com/rest/reference/reactions#list-reactions-for-a-pull-request-review-comment"
}
</code></pre>
<p>Am I using the incorrect? I’m not sure what I’m missing.</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/myyk">@myyk</a>,</p>
<p>Glad to see you in Github Community Forum!</p>
<p>The API url should be  “/repos/{owner}/{repo}/<strong>issues</strong>/comments/{comment_id}/reactions”, in your curl  command, it’s ‘pulls’ which is not correct.</p>
<p>In your action index.js <a href="https://github.com/myyk/git-democracy/blob/536c0525895f8bea32e083baa4a3e97d032238ff/.github/actions/tally-javascript-action/index.js#L13" rel="nofollow noopener">line 13</a>, please add <strong>issues</strong> as well, should be:</p>
<pre><code class="lang-auto">await octokit.request('GET /repos/{owner}/{repo}/issues/comments/{comment_id}/reactions', {
</code></pre>
<p>Thanks</p>

**Comment by myyk** [comment]
<p>Hi <a class="mention" href="/weide-zhou">@weide-zhou</a> ,</p>
<p>Thank you for your response! Sorry for the confusing link above, the docs should be at <a href="https://docs.github.com/en/rest/reference/reactions#list-reactions-for-a-pull-request-review-comment" rel="nofollow noopener">https://docs.github.com/en/rest/reference/reactions#list-reactions-for-a-pull-request-review-comment</a> since I am in trying to list reactions on a pull request comment (not on an issue’s comment as the link was).</p>
<p>I must be using the API incorrectly. I think if we solve why the curl request is failing it might become obvious what’s wrong in code.</p>
<p><code>curl -H "Accept: application/vnd.github.squirrel-girl-preview+json" https://api.github.com/repos/myyk/git-democracy/pulls/comments/677573350/reactions</code></p>
<p>This returns <code>"Not Found"</code> when I think it should be pulling data from <a href="https://github.com/myyk/git-democracy/pull/10#issuecomment-677573350" rel="nofollow noopener">https://github.com/myyk/git-democracy/pull/10#issuecomment-677573350</a>.</p>
<p>Do you see anything wrong here?</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/myyk">@myyk</a>,</p>
<p>Thanks for your quick reply!</p>
<p>pull request is actually an issue.</p>
<p>The comment on your pull request is not a review comment, you can confirm with rest api  <a href="https://developer.github.com/v3/pulls/comments/#list-review-comments-on-a-pull-request" rel="nofollow noopener">here</a>.</p>
<p>Hence, you should use the api below for comment reaction:<br>
<a href="https://docs.github.com/en/rest/reference/reactions#list-reactions-for-an-issue-comment" class="onebox" target="_blank" rel="nofollow noopener">https://docs.github.com/en/rest/reference/reactions#list-reactions-for-an-issue-comment</a></p>
<pre><code class="lang-auto">GET /repos/{owner}/{repo}/issues/comments/{comment_id}/reactions
</code></pre>

**Comment by myyk** [comment]
<p>Hi <a class="mention" href="/weide-zhou">@weide-zhou</a>,</p>
<p>Oh yeah, those are do use that different API, thanks for helping me understand that github nomenclature.</p>
<p>I have a small follow up question that. In the API docs, I see this:</p>
<pre><code class="lang-auto">The `reactions` key will have the following payload where url can be used to construct the API location for listing and creating reactions.

{
  "total_count": 5,
  "+1": 3,
  "-1": 1,
  "laugh": 0,
  "confused": 0,
  "heart": 1,
  "hooray": 0,
  "url": "https://api.github.com/repos/octocat/Hello-World/issues/1347/reactions"
}
</code></pre>
<p>But in my request, I don’t a <code>reactions</code> section as described. Is there something I’m not passing in the call or about the way it’s called that’s not returning it?</p>
<pre><code class="lang-auto">curl  \
 -H "Accept: application/vnd.github.squirrel-girl-preview+json"   \
  https://api.github.com/repos/myyk/git-democracy/issues/comments/677573350/reactions
</code></pre>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/myyk">@myyk</a>,</p>
<p>As the API doc mentioned, this reaction object exists in issue coment payload, you can get it via list the issue comment with media type: <code>application/vnd.github.squirrel-girl-preview</code>.<br>
</p><aside class="onebox allowlistedgeneric">
  <header class="source">
      <a href="https://developer.github.com/v3/issues/comments/#get-an-issue-comment" target="_blank" rel="nofollow noopener">GitHub Developer</a>
  </header>
  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/52772761/181095020-ac89e581-f9f4-461b-8dbc-cdc24b5212ae.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/2/26361d706a8a255479f0cb328577f99ad8731275_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/2/26361d706a8a255479f0cb328577f99ad8731275_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/2/26361d706a8a255479f0cb328577f99ad8731275_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/2/26361d706a8a255479f0cb328577f99ad8731275_2_10x10.png">

<h3><a href="https://developer.github.com/v3/issues/comments/#get-an-issue-comment" target="_blank" rel="nofollow noopener">Issue Comments</a></h3>

<p>Get started with one of our guides, or jump straight into the API documentation.</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>
<br>
<div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/2/2ee5e44d674e0d840d1ab55ff43fc768fca3c78d.png" data-download-href="/uploads/short-url/6GSvhxDxOCCtDEJnHpnMWxrGYR7.png?dl=1" title="image" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/52772761/181095021-bc9c5b3d-dc5c-4bda-9095-64cd47b43cfa.png" alt="image" data-base62-sha1="6GSvhxDxOCCtDEJnHpnMWxrGYR7" width="690" height="477" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/2/2ee5e44d674e0d840d1ab55ff43fc768fca3c78d_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">841×582 28.2 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Thanks</p>

**Comment by myyk** [comment, marked as answer]
<p>Thank you <a class="mention" href="/weide-zhou">@weide-zhou</a>! I was able to make a request exactly as I wanted to. Thank you for your help!</p>
<p>To summarize what you showed me in regard to getting reaction summary info from <a href="https://github.com/myyk/git-democracy/pull/10#issuecomment-677573350:" rel="nofollow noopener">https://github.com/myyk/git-democracy/pull/10#issuecomment-677573350:</a></p>
<ol>
<li>
<p>I was using the wrong API since the comment uses the issue comments API, not the pull request comments API even though it appears on the Pull Request page. <a href="https://developer.github.com/v3/issues/comments/#get-an-issue-comment" rel="nofollow noopener">https://developer.github.com/v3/issues/comments/#get-an-issue-comment</a></p>
</li>
<li>
<p>I needed to add the <code>Accept: application/vnd.github.squirrel-girl-preview</code> header.</p>
</li>
</ol>
<p>I ended up with something like this:</p>
<pre><code class="lang-auto">const octokit = github.getOctokit(
  inputs.token,
  {
    previews: ['squirrel-girl'],
  },
);

octokit.issues.getComment({
  owner: owner,
  repo: repo,
  comment_id: commentId,
});
</code></pre>
